### ISSUE TITLE
Initializing non-nullable property to itself or another uninitialized property should warn

### ISSUE THREAD

**Issue by davidmilligan** [dotnet/roslyn#37905]
**Version Used**: 
16.3.0-preview-19329-01+d31fdbf01

**Steps to Reproduce**:

```
#nullable enable
class A { }
class B
{
    A C { get; set; }
    B()
    {
        C = C;
    }
}
```

**Expected Behavior**:

Compiler warning that property 'C' is not initialized

**Actual Behavior**:

No compiler warning

---

Of course the above is a trivial example, but there are much more sinister possibilities:
```
A C1 { get; set; }
A C2 { get; set; }
B(A c2)
{
  C1 = C2; //oops, that was supposed to be c2
  C2 = C1;
}

**Comment by RikkiGibson** [comment]
I think this pretty much falls in the warning waves bucket. Please correct me if I'm wrong.

**Comment by davidwengier** [comment]
:+1:
I ran into this yesterday and was surprised that I didn't get a Nullable warning here.

**Comment by RikkiGibson** [comment]
It seems like there's actually 2 ways to solve this particular scenario:

1. We can decide that assigning a property to itself is almost certainly a bug that we should warn about (in which case this is a warning waves issue)
2. We can decide that non-nullable properties and fields in constructors should have an initial flow state of maybe-null (in which case this is a nullable issue)

I'm curious what @cston and @jaredpar think about this. I would actually like to do both but perhaps there are reasons we have decided not to do (2).

**Comment by RikkiGibson** [comment]
This now reports `warning CS8601: Possible null reference assignment` on the assignment. This is because auto-properties now have maybe-null initial state in constructors.
